crates/matrix-sdk/src/client/mod.rs

```diff
@@ -1691,12 +1691,24 @@ impl Client {
     /// assert!(client.create_room(request).await.is_ok());
     /// # });
     /// ```
-    pub async fn create_room(&self, request: create_room::v3::Request) -> HttpResult<room::Joined> {
+    pub async fn create_room(&self, request: create_room::v3::Request) -> Result<room::Joined> {
```

Comment:
The idea here was to have a single method (bindings side) for both dm and normal rooms and let Rust to decide how to build the final ruma request.
If we think this approach isn't ideal I can expose a second method `create_dm_room` to UniFFI.
What do you think?